### DISCUSSION TITLE
About the parameter settings for ‘ps_options’ and ‘phase_linking’

### DISCUSSION THREAD

**Discussion by ichaochao9** [isce-framework/dolphin#398] [Q&A]
Hello developers,

I have recently been processing InSAR data using 'dolphin', focusing on a mountainous area with complex terrain. I am hoping to utilize 'dolphin' to select appropriate PS and DS points. However, the results I obtained are far from satisfactory, with noticeable unwrapping errors. I would like to seek your advice on how to set parameters such as 'ps_options', 'phase_linking', and 'unwrap_options' when processing mountainous data with 'dolphin'.

**Comment by scottstanie** [comment]
Hi, thanks for testing it out-
Mountainous areas are known to be difficult, but I think we'd need to see a little more detail about what was tried and what "far from satisfactory" means.
What input SLCs did you use/ what unwrapping did you try?

**Reply by ichaochao9** [reply]
Hello Doctor,

I have used SLCs from ALOS2 data (processed with GAMMA) and Sentinel-1 data (processed with ISCE2) as inputs. For unwrapping methods, I have tested the 'spurt' method in dolphin, and the unwrapping results have noticeable unwrapping errors in many areas as shown in the figure below. In the setting of PS/DS and ‘spurt’ parameters, I mostly chose the default values.
![图1](https://github.com/user-attachments/assets/eca20a3e-b23d-4d40-9c8e-7281eda87ba6)
![图2](https://github.com/user-attachments/assets/ce77a919-8abb-4249-b2a3-34af0fe5b853)

**Comment by ichaochao9** [comment]
Hello developers,

Could you please review the attached 'dolphin_config.yaml' file for any improper parameter settings? I've also included images of the 'shp_counts' and an unwrapping result from Sentinel-1 data, which show errors in the tiling process.
[dolphin_config.docx](https://github.com/user-attachments/files/16762177/dolphin_config.docx)


<img width="653" alt="image" src="https://github.com/user-attachments/assets/e8d2d972-0c05-4a42-85de-db6973a947a6">
<img width="1004" alt="image" src="https://github.com/user-attachments/assets/bf2c6812-7f7a-4513-a174-cfa12c5fd996">

**Reply by scottstanie** [reply]
Interesting, that colorbar is shrunk to show the tiling artifact, right?

So I think this may still be an issue with spurt. Do you know how many tiles spurt used? The location of the tiles is weird, but it tries to use a density scheme

**Reply by ichaochao9** [reply]
Yes, Doctor. I have zoomed in on the image to show the discontinuity at the tile junctions. But I did not shrink the colorbar, it was generated by default. Upon checking, I found that spurt used 9 tiles. The following image is the unwrapped result of other interferogram.
<img width="527" alt="image" src="https://github.com/user-attachments/assets/1f329be8-4abb-4a6f-b2b6-3b7cb5fb9fe0">

**Comment by scottstanie** [comment]
I took another look at the full `dolphin_config.yaml` and noticed a couple things:
1. you have "max_temporal_baseline: 3". This means "include interferograms with a maximum of 3 days temporal baseline". This is probably not what you want, and added 0 new interferograms to the network.
2. You have both `reference_index: 85` and `max_bandwidth: 3`. I would expect this to lead to the [following error if you try to run spurt](https://github.com/isce-framework/dolphin/blob/main/src/dolphin/unwrap/_unwrap_3d.py#L96-L98):

```
ValueError: interferograms for spurt must be single reference.
```

I realized it's confusing now (spurt is very new and under active developement; this means the dolphin interface to spurt must be even more in flux), but you pass single-reference phase linked interferograms to spurt, then it forms it's own Hope-3 network. This can/will change later, but it was the first working version.

Did you change the config file/run spurt manually to get this?
